crd-generator/api-v2/src/main/java/io/fabric8/crdv2/generator/AbstractCustomResourceHandler.java

```diff
@@ -44,29 +46,35 @@ void addPrinterColumn(String path, String column, String format,
 
   protected void handlePrinterColumns(AbstractJsonSchema<?, ?> resolver, PrinterColumnHandler handler) {
     TreeMap<String, AnnotationMetadata> sortedCols = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
+    resolver.getAdditionalPrinterColumns().forEach(apc -> sortedCols.put(apc.path(), new AnnotationMetadata(apc, null)));
     sortedCols.putAll(resolver.getAllPaths(PrinterColumn.class));
     sortedCols.forEach((path, property) -> {
-      PrinterColumn printerColumn = ((PrinterColumn) property.annotation);
-      String column = printerColumn.name();
-      if (Utils.isNullOrEmpty(column)) {
-        column = path.substring(path.lastIndexOf(".") + 1).toUpperCase();
-      }
-      String format = printerColumn.format();
-      format = Utils.isNotNullOrEmpty(format) ? format : null;
-      int priority = printerColumn.priority();
-      String type = property.schema.getType();
-      if ("object".equals(type) || "array".equals(type)) {
-        LOGGER.warn("Printer column '{}' has a type '{}' that is not allowed, will use string intead", column, type);
-        type = "string";
-      } else if ("string".equals(type) && "date".equals(property.schema.getFormat())) {
-        type = "date";
-      }
+      if (property.annotation instanceof AdditionalPrinterColumn) {
```

Comment:
At the moment the `@AdditionalPrinterColum` is collected from everywhere in the spec or status type. Is this intended?

We should either clarify this in the description of the annotation or pick it only from the custom resource class.

Reply:
> At the moment the @AdditionalPrinterColum is collected from everywhere in the spec or status type.

It should not be: https://github.com/fabric8io/kubernetes-client/pull/6390/files#diff-e5491c474236ab12f8e6f5d9895464ecf77754b17ddb0e182682add6db8620ceR146

> We should either clarify this in the description of the annotation or pick it only from the custom resource class.

It is documented as intended for the root: https://github.com/fabric8io/kubernetes-client/pull/6390/files#diff-669d381122146d717c360f17fcfe5eeaab88b5c84dd4bfcf81b6de71aa5a75c1R25

Reply:
Sorry, seems like it was a problem on my side. I have checked out your branch again and could not reproduce it. Looks fine now!